### PR TITLE
Mega refactor for Person

### LINKED ISSUE #138: Refactor all person instances to buyer

### PR DESCRIPTION
**Pull request by riccqi**
Closes #138 
Change all tests and mentions of person to buyer

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/EditBuyerCommand.java [hunk 1]
```diff
@@ -57,8 +57,8 @@ public class EditBuyerCommand extends Command {
     private final EditPersonDescriptor editPersonDescriptor;
 
     /**
-     * @param index of the person in the filtered person list to edit
-     * @param editPersonDescriptor details to edit the person with
+     * @param index                of the buyer in the filtered buyer list to edit
+     * @param editPersonDescriptor details to edit the buyer with
      */
     public EditBuyerCommand(Index index, EditPersonDescriptor editPersonDescriptor) {
         requireNonNull(index);
```
Comment (jchilling) [inline src/main/java/seedu/address/logic/commands/EditBuyerCommand.java:64]: Should the `EditPersonDescriptor` and its methods be refactored also

File: src/main/java/seedu/address/model/buyer/UniquePersonList.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    private boolean personsAreUnique(List<Buyer> buyers) {
+        for (int i = 0; i < buyers.size() - 1; i++) {
+            for (int j = i + 1; j < buyers.size(); j++) {
+                if (buyers.get(i).isSamePerson(buyers.get(j))) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
```
Comment (jchilling) [inline src/main/java/seedu/address/model/buyer/UniquePersonList.java:136]: Method name is persons but compares buyers. Actually, do we want to consider adding a parent class `Person` and a `Buyer` that extends it

**Review by jchilling — APPROVED** [review]
LGTM 👍 Do we want to consider making `Person` a parent class with methods like isSamePerson etc. and `Buyer` extends it? We can mention in the DG that this way it is easier for extensions in the future. For instance, a `Seller` class can be added and still share PersonBuilder, getter methods, etc.

File: LICENSE [hunk 3]
```diff
@@ -2,11 +2,11 @@ MIT License
...
 Copyright (c) 2016 Software Engineering Education - FOSS Resources
 
-Permission is hereby granted, free of charge, to any person obtaining a copy
+Permission is hereby granted, free of charge, to any buyer obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
-copies of the Software, and to permit persons to whom the Software is
+copies of the Software, and to permit buyers to whom the Software is
 furnished to do so, subject to the following conditions:
```
[the remark this answers is not in the document]
Comment (riccqi) [reply LICENSE:11]: LOL good catch

File: src/main/java/seedu/address/model/buyer/UniquePersonList.java (hunk 2 above)
Comment (riccqi) [reply src/main/java/seedu/address/model/buyer/UniquePersonList.java]: yea makes sense, we can KIV for the purpose of better OOP

Comment (riccqi) [reply src/main/java/seedu/address/model/buyer/UniquePersonList.java]: created issue #141

**Pull request merged**